applications/system/js_app/examples/apps/Scripts/Examples/i2c.js

```diff
@@ -0,0 +1,45 @@
+// Connect an 24C32N EEPROM to the I2C bus of the board. SDA=pin 15, SCL=pin 16, VCC=pin 9, GND=pin 8.
+let i2c = require("i2c");
+
+function i2c_find_first_device() {
+    let addr = -1;
+    for (let try_addr = 0; try_addr !== 0xff; try_addr++) {
+        if (i2c.isDeviceReady(try_addr, 5)) {
+            addr = try_addr;
+            break;
+        }
+    }
+    return addr;
+}
+
+let addr = i2c_find_first_device();
+if (addr === -1) {
+    print("I2C device not found");
+    print("Please connect a 24C32N  EEPROM I2C device to the Flipper Zero.");
+    print("SDA=pin 15, SCL=pin 16,  VCC=pin 9, GND=pin 8.");
+} else {
+    print("I2C device found at address: " + to_hex_string(addr));
+    delay(1000);
+
+    // first two bytes are the start address (0x0000)
+    // the remaining bytes are the data to store.
+    i2c.write(addr, [0x00, 0x00, 0x41, 0x42, 0x43, 0x44, 0x45, 0x46, 0x47]);
+    while (i2c.isDeviceReady(addr, 9) === false) {
+        print("Waiting for device to be ready...");
+    }
+
+    // write the address to read from (we start at address 0x0001)
+    // read 3 bytes - 0x42, 0x43, 0x44
+    let data = i2c.writeRead(addr, [0x00, 0x01], 3, 100);
```

Comment:
function name isnt super clear (and neither is the internal `furi_hal_i2c_trx()`) but from what i can see its the same as doing a write followed by a read. so then, the comment might need some more context. ive never used i2c, but from this code i think i understand that in any tx the first 2 bytes are used to move a memory cursor, so a tx with more than 2 bytes will move the cursor then start writing there, while a tx with just 2 bytes is only moving the cursor, so then the read can proceed at the right position?